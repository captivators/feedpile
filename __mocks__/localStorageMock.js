const localStorageMock = (() => {
  let store = {
    "profile": '{"name":"faiz"}',
    id_token: ""
  };
  return {
    getItem(key) {
      return store[key]
    },
    setItem(key, value) {
      store[key] = value.toString()
    },
    clear() {
      store = {}
    }
  };
})();

global.localStorage = localStorageMock;
