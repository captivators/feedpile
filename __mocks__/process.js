const process = {
  "env" : {
    "AUTH_ID":"test",
    "AUTH_DOMAIN":"user_id"
  }
}
global.process = process;
