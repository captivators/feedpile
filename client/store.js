import {createStore, compose, applyMiddleware} from 'redux';
import rootReducer from './root-reducer';
import logger from 'redux-logger';
import createSagaMiddleware from 'redux-saga';
import rootSaga from './root-saga';

const sagaMiddleware = createSagaMiddleware();

const store = createStore(rootReducer, compose (
  applyMiddleware(logger, sagaMiddleware)
));

sagaMiddleware.run(rootSaga);
export default store;
