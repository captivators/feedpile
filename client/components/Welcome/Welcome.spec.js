import '../../../__mocks__/localStorageMock'
import React from 'react';
import { shallow, render } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';
import Welcome from './Welcome'
import store from '../../store';
import { Provider } from 'react-redux';

test('Welcome take a snapshot', () => {
  const component = shallow(<Welcome store={ store }/>);
  const tree = shallowToJson(component);
  expect(tree).toMatchSnapshot();
});

test('Welcome renders welcome image', () => {
  const component = render(<MuiThemeProvider><Provider store={ store }><Welcome /></Provider></MuiThemeProvider>)
  expect(component.find('.newspaper').length).toEqual(1)
  expect(component.find('.welcome-container').length).toEqual(1)
})