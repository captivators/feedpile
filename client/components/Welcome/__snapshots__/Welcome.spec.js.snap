// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Welcome take a snapshot 1`] = `
<Welcome
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
  toggleModal={[Function]}
/>
`;
